target/linux/ath79/dts/qca9558_comfast_cf-wr650ac.dtsi

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca9557.dtsi"
+
+/ {
+	chosen {
+                bootargs = "console=ttyS0,115200n8";
+        };
+
+	leds {
+		compatible = "gpio-leds";
+
+		wps {
+			label = "comfast:blue:wps";
+			gpios = <&gpio 20 GPIO_ACTIVE_LOW>;
+		};
+
+		network {
+			label = "comfast:blue:network";
+			gpios = <&gpio 4 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan2g {
+			label = "comfast:blue:24g";
+			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy1tpt";
+		};
+
+		wlan5g {
+			label = "comfast:blue:58g";
+			gpios = <&gpio 2 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <100>;
```

Comment:
You may try using gpio-keys and removing poll-interval here.

Reply:
may issues is the WAN.
poll-interval and all the rest works fine.

Do you have any insights on why WAN is not recognised in the switch ?

Reply:
I will have a look later, but I'm not an expert with this stuff. However, a few weeks ago there was a patch changing eth0/eth1 order, this might have had an effect, too.

Reply:
fixed